client/blocks/site-preview/index.jsx

```diff
@@ -1,110 +1,82 @@
-import debugFactory from 'debug';
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import WebPreview from 'calypso/components/web-preview';
 import { addQueryArgs } from 'calypso/lib/route';
 import isDomainOnlySite from 'calypso/state/selectors/is-domain-only-site';
 import { getSiteOption, getSiteSlug, isSitePreviewable } from 'calypso/state/sites/selectors';
 import { getCurrentLayoutFocus } from 'calypso/state/ui/layout-focus/selectors';
 import { closePreview } from 'calypso/state/ui/preview/actions';
-import {
-	getPreviewSite,
-	getPreviewSiteId,
-	getPreviewUrl,
-} from 'calypso/state/ui/preview/selectors';
+import { getPreviewSiteId, getPreviewUrl } from 'calypso/state/ui/preview/selectors';
 
-const debug = debugFactory( 'calypso:site-preview' );
+function usePreviewCounter( siteId, showPreview ) {
+	const [ previewCount, setPreviewCount ] = useState( 0 );
 
-class SitePreview extends Component {
-	static propTypes = {
-		className: PropTypes.string,
-		showPreview: PropTypes.bool,
-		previewUrl: PropTypes.string,
-		selectedSite: PropTypes.object,
-		selectedSiteId: PropTypes.number,
-		selectedSiteNonce: PropTypes.string,
-		selectedSiteUrl: PropTypes.string,
-		closePreview: PropTypes.func.isRequired,
-	};
+	useEffect( () => {
+		if ( siteId ) {
+			setPreviewCount( 0 );
+		}
+	}, [ siteId ] );
 
-	state = {
-		previewCount: 0,
-	};
+	useEffect( () => {
+		if ( showPreview ) {
+			setPreviewCount( ( n ) => n + 1 );
+		}
+	}, [ showPreview ] );
```

Comment:
The trouble with both `useEffect` calls is that the values used for render can get out of sync. When `siteId` (or `previewUrl`) changes, or when `showPreview` changes, there are two renders: one with old value of `previewCount`, one with new value of `previewCount`.